ecosystem/sep-0038.md

```diff
@@ -0,0 +1,176 @@
+
+`delivery_methods` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`name` | string | The value to use when making `POST /quote` requests.
+`description` | string | A human readable description of the method identified by `name`.
+
+### GET Prices
+
+This endpoint can be used to fetch the [indicative](https://www.investopedia.com/terms/i/indicativequote.asp) prices of available off-chain assets in exchange for a Stellar asset and vice versa.
+
+#### Request
+
+Only one of the following parameters should be included in the request.
+
+Name | Type | Description
+-----|------|------------
+`asset` | string | (optional) The [SEP-32](sep-0032.md) address of the Stellar asset you want to get the price of available off-chain currencies in. Lumens should be specified as `XLM`.
+`currency` | string | (optional) The [ISO 4217](https://www.iso.org/iso-4217-currency-codes.html) code of the currency you want to get the price of available Stellar assets in.
+
+#### Response
+
+Only one of the following attributes should be included in the response.
+
+Name | Type | Description
+-----|------|------------
+`assets` | array | (optional) An array of objects containing the available Stellar assets and their prices in terms of the `currency` provided in the request.
+`currencies` | array | (optional) An array of objects containing the available currencies and their prices in terms of the `asset` provided in the request.
+
+`assets` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`asset` | string | Same as the definition of `asset` in the request.
+`price` | string | The price of one unit of `asset` in terms the `currency` sent in the request.
+`decimals` | integer | The number of decimals needed to represent asset.
+
+`currencies` Object Schema
+
+Name | Type | Description
+-----|------|------------
+`currency` | string | Same as the definition of `currency` in the request.
+`price` | string | The price of one unit of `currency` in terms of the `asset` sent in the request.
+`decimals` | integer | The number of decimals needed to represent the currency.
+
+### GET Price
+
+This endpoint can be used to fetch the indicative price of a Stellar asset and currency [pair](https://en.wikipedia.org/wiki/Currency_pair).
+
+#### Request
+
+Name | Type | Description
+-----|------|------------
+`base` | string | The base asset or currency. Ex. `USDC:circle.io`, `ARS`
+`counter` | string | The counter asset or currency.
+`base_amount` | string | The amount of the base asset or currency to be traded for the counter asset or currency.
+
+#### Response
+
+Name | Type | Description
+-----|------|------------
+`price` | string | The price of one unit of the base asset or currency in terms of the counter asset or currency for the `base_amount` specified in the request.
```

Comment:
❔  I want to make sure I understand how these three fields will be used, specially the amount. If I request `base=USDC`, `counter=USDC2`, where `USDC/USDC2 = 2` (i.e. `1 USDC = 2 USDC2`), with `base_amount=50`, I should expect `price=200`?

Reply:
To be clear, this API is not built for exchanging two stellar assets. There may be value in such a standard, but the request/response schemas would be fairly different. For example, we could returned signed transactions consisting of two payment operations with the quote expiration as the `maxTime`.

Maybe we should make sure we name this protocol something that doesn't conflict with such a standard in the future?

Reply:
But to answer your question, this `base/counter` syntax has a specific [definition](https://www.investopedia.com/terms/c/currencypair.asp):

> The quotation EUR/USD = 1.2500 means that one euro is exchanged for 1.2500 U.S. dollars. In this case, EUR is the base currency and USD is the quote currency (counter currency). This means that 1 euro can be exchanged for 1.25 U.S. dollars. Another way of looking at this is that it will cost you $125 to buy 100 euros.

Reply:
Thanks, that's what I wanted to confirm that we are using that definition. We made some mistakes in the past where we mixed up counter and base in the ticket API and just wanted to check. 

Also +1 you're right price would be 100, I messed that up. 😄.

I don't think we need to worry too much about stellar asset to stellar asset. I was just using them as and sample, a bad example given that's not what this api is for.

Reply:
I'm un-resolving this thread because I think I was misunderstanding `base` and `counter`.

For a given currency pair, say `EUR/USD`, there are two parties involved. One party is buying EUR and selling USD, and the other is selling EUR and buying USD. `EUR/USD = 1.25` simply means 1 unit of `EUR` is equal to 1.25 units of USD. 

This syntax does _not_ indicate which party is selling the base currency and buying the counter currency, but this API assumes its the client. I think this can be confusing for people who understand what `base/counter` syntax actually is.

The 0x API uses `sellToken`/`sellAmount` and `buyToken`/`buyAmount`, written in the client's perspective. I think we should use this pattern as well. (`sellAsset`/`sellAmount`) Thoughts?

Reply:
👍🏻 I think being explicit about the sell and buy tokens would be a great idea. In that case I also wonder if we should return not only a price, but a total amount.

Reply:
It can't hurt 👍